source/contributing/implementation/codestyle.rst

```diff
@@ -0,0 +1,153 @@
+
+  * File headers must contain the license headers for the project. Use the ``licenseFormat`` Gradle task to add them
+    automatically.
+
+* Imports
+
+  * Imports must be grouped in the following order, where each group is separated by an empty line
+
+    * Static imports
+    * All other imports
+    * ``java`` imports
+    * ``javax`` imports
+
+  * This differs from Google's style in that imports are not grouped by top-level package, they are all grouped as one.
+
+* Exceptions
+
+  * For exceptions that are to be ignored, name the exception variable ``ignored``
+
+* Field accesses
+
+  * Qualify **all** field accesses with ``this``
+
+* Javadocs
+
+  * Do not use ``@author``
+  * Wrap additional paragraphs in ``<p>`` and ``</p>``
+  * Capitalize the first letter in the descriptions within each "at clause", i.e. ``@param name Player to affect``, no
+    periods
+
+Code Conventions
+================
+
+* Use `Optionals <https://code.google.com/p/guava-libraries/wiki/UsingAndAvoidingNullExplained>`_ instead of returning
+  ``null`` in the API
+* Method parameters accepting ``null`` must be annotated with ``@Nullable`` (from javax.*), all methods and parameters
+  are ``@Nonnull`` by default.
+* Use `Google Preconditions <https://code.google.com/p/guava-libraries/wiki/PreconditionsExplained>`_ for null- and
+  argument checking.
+
+The Gist
+========
+
+While we urge that you read Google's Java conventions particularly, the two are fairly long documents. To get you
+started quickly, here is an example of properly formatted code:
+
+.. code-block:: java
+
+    /*
+     * This file is part of Sponge, licensed under the MIT License (MIT).
+     *
+     * Copyright (c) SpongePowered.org <http://www.spongepowered.org>
+     * Copyright (c) contributors
+     *
+     * Permission is hereby granted, free of charge, to any person obtaining a copy
+     * of this software and associated documentation files (the "Software"), to deal
+     * in the Software without restriction, including without limitation the rights
+     * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+     * copies of the Software, and to permit persons to whom the Software is
+     * furnished to do so, subject to the following conditions:
+     *
+     * The above copyright notice and this permission notice shall be included in
+     * all copies or substantial portions of the Software.
+     *
+     * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+     * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+     * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+     * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+     * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+     * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+     * THE SOFTWARE.
+     */
+
+    package com.example.java;
```

Comment:
In most of the codebase, there is no newline between the license header and package.